test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1049,3 +1069,30 @@
                         (utils/get-extension-value utils/subject-alt-name-oid)
                         (:dns-name))]
       (is (= #{"puppet" "master"} (set alt-names))))))
+
+(deftest file-permissions
+  (let [tmpdir (System/getProperty "java.io.tmpdir")]
+    (testing "A newly created file contains the properly set permissions"
+      (doseq [u all-perms
+              g no-write-perms
```

Comment:
My guess is that this is due to a more restrictive default umask causing the attempt to set the write bit for g and o to not be honored for the `Files/createFile` call.  From some text about `Files/createFile` in https://docs.oracle.com/javase/tutorial/essential/io/fileAttr.html#posix:

> Note that the umask also applies, so the new file might be more secure than the permissions that were requested.

I've found that if I turn around and call `Files/setPosixFilesPermissions` on the newly created file with a file permission mask that sets the write bit for g and o that the full mask is honored properly.  I modified the implementation of `create-file-with-perms` to:

~~~~clojure
(schema/defn create-file-with-perms :- File
 ...
  [path :- schema/Str
   permissions :- schema/Str]
  (let [permissions-set (PosixFilePermissions/fromString permissions)
        path-obj (-> (get-path-obj path)
                     (Files/createFile
                      (into-array FileAttribute
                                  [(PosixFilePermissions/asFileAttribute
                                    permissions-set)])))]
    (Files/setPosixFilePermissions path-obj permissions-set)
    (.toFile path-obj)))
~~~~

With the above and `all-perms` used both for g and o in both of the `file-permissions` tests, the tests all pass for me on my OS X laptop.  I think it would be good to add something like this to ensure that the *exact* permissions we're intending to set are applied for the newly created file.